### PR TITLE
Split build rule in lib/BUILD.

### PR DESCRIPTION
**Pull request by sergeisakov**

### CODE REVIEW

**Comment by 95-martin-orion** [comment]
Test failures appear to be a side-effect of the Cirq v0.8 release - I'll ignore them in my review.

File: lib/BUILD [hunk 1]
```diff
@@ -1,33 +1,233 @@
...
+cc_library(
+    name = "fuser",
+    hdrs= [
         "fuser.h",
+    ],
+)
+
+cc_library(
+    name = "io",
+    hdrs= [
```
[the remark this answers is not in the document]
Comment (sergeisakov) [reply lib/BUILD]: Done.

File: lib/BUILD [hunk 2]
```diff
@@ -1,33 +1,233 @@
 package(default_visibility = ["//visibility:public"])
 
-# TODO: split into (AVX|SSE|basic) libraries.
 cc_library(
-    name = "qsim_lib",
```
[the remark this answers is not in the document]
Comment (sergeisakov) [reply lib/BUILD:5]: Done. Note that the aggregate libraries I added are not suitable for TFQ.

File: lib/BUILD [hunk 3]
```diff
@@ -1,33 +1,233 @@
...
+    hdrs= [
         "gates_appl.h",
+    ],
+    deps = [
+        ":matrix",
+    ],
+)
+
+cc_library(
+    name = "gates_qsim",
```
[the remark this answers is not in the document]
Comment (sergeisakov) [reply lib/BUILD:127]: Added some comments.

**Pull request merged**